### PR TITLE
feat: query earliest attestation nonce

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

So that the orchestartor is able to query the earliest attestation nonce and sign up to it.

This change is not breaking and is just adding a new API endpoint.

PS: I originally thought that this was already implemented. But sadly, we didn't.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: proto/celestia/qgb/v1/query.proto [hunk 1]
```diff
@@ -30,6 +30,11 @@ service Query {
       returns (QueryLatestAttestationNonceResponse) {
     option (google.api.http).get = "/qgb/v1/attestations/nonce/latest";
   }
+  // LatestAttestationNonce queries latest attestation nonce.
```
Comment (rootulp) [inline proto/celestia/qgb/v1/query.proto:33]:
```suggestion
  // EarliestAttestationNonce queries the earliest attestation nonce.
```

File: proto/celestia/qgb/v1/query.proto [hunk 2]
```diff
@@ -30,6 +30,11 @@ service Query {
...
+  rpc EarliestAttestationNonce(QueryEarliestAttestationNonceRequest)
+      returns (QueryEarliestAttestationNonceResponse) {
+    option (google.api.http).get = "/qgb/v1/attestations/nonce/earliest";
```
Comment (rootulp) [inline proto/celestia/qgb/v1/query.proto:36]: [question] did the rename PR not include a rename to the endpoint?

I would expect `/blockstream/v1` and not `/qgb/v1/

Comment (SweeXordious) [reply proto/celestia/qgb/v1/query.proto:36]: that would be breaking, we wanted to keep that until v2

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

File: proto/celestia/qgb/v1/query.proto (hunk 2 above)
Comment (evan-forbes) [reply proto/celestia/qgb/v1/query.proto:36]: to clarify the smallest bit, we want to add the consensus breaking things, but we first need the infra to be able to test the upgrade/migrations

Also, if we don't already have an issue for that, can we get one?

Comment (SweeXordious) [reply proto/celestia/qgb/v1/query.proto:36]: https://github.com/celestiaorg/celestia-app/issues/2733

**Review by evan-forbes — APPROVED** [review]
I think I saw these changes in the rename PR as well

But LGTM! 👍

**Pull request merged**